components/pd_client/src/lib.rs

```diff
@@ -43,6 +43,37 @@ pub use self::{
 pub type Key = Vec<u8>;
 pub type PdFuture<T> = BoxFuture<'static, Result<T>>;
 
+#[derive(Clone, Debug, Default)]
+pub struct RegionWriteCfCopDetail {
+    pub next: usize,
+    pub prev: usize,
+    pub processed_keys: usize,
+}
+
+impl RegionWriteCfCopDetail {
+    pub fn new(next: usize, prev: usize, processed_keys: usize) -> Self {
+        Self {
+            next,
+            prev,
+            processed_keys,
+        }
+    }
+
+    pub fn add(&mut self, other: &RegionWriteCfCopDetail) {
+        self.next += other.next;
+        self.prev += other.prev;
+        self.processed_keys += other.processed_keys;
+    }
+
+    pub fn sub(&self, other: &RegionWriteCfCopDetail) -> Self {
+        Self::new(
+            self.next - other.next,
+            self.prev - other.prev,
+            self.processed_keys - other.processed_keys,
+        )
+    }
+}
+
 #[derive(Default, Clone, Debug)]
 pub struct RegionStat {
```

Comment:
It's not a good place for `RegionWriteCfCopDetail`, because it dedicated for region heartbeat, all its fields are reported to PD.

Reply:
1. I think this info can be a info used by PD. 2. region collector collects the info from region heartbeat, so it has to be added in RegionStat to make region collector collects it.

Reply:
Okay, please move the `cop_detail` to the last, and add some comments that what it is and why it is not reported to PD.